clients/upslog.c

```diff
@@ -393,28 +414,53 @@ static void run_flist(void)
 	 * -u <username>
 	 */
 
+static void term_handler(int signo)
+{
+	if (signo != SIGCHLD && monhost_child_anchor != NULL) {
+		for (monhost_child_current = monhost_child_anchor;
+		     monhost_child_current != NULL;
+		     monhost_child_current = monhost_child_current->next)
+			kill(monhost_child_current->pid, signo);
+		fatalx(EXIT_FAILURE, "Killed by user");
+	}
+}
+
 int main(int argc, char **argv)
 {
 	int	interval = 30, i, foreground = -1;
+	size_t	monhost_len;
 	const char	*prog = xbasename(argv[0]);
 	time_t	now, nextpoll = 0;
 	const char	*user = NULL;
 	struct passwd	*new_uid = NULL;
-	const char	*pidfilebase = prog;
+	char	*pidfilebase = prog;
 
 	logformat = DEFAULT_LOGFORMAT;
 	user = RUN_AS_USER;
 
 	printf("Network UPS Tools %s %s\n", prog, UPS_VERSION);
 
-	while ((i = getopt(argc, argv, "+hs:l:i:f:u:Vp:FB")) != -1) {
+	while ((i = getopt(argc, argv, "+hs:l:i:f:u:Vp:FBm:")) != -1) {
 		switch(i) {
 			case 'h':
 				help(prog);
 #ifndef HAVE___ATTRIBUTE__NORETURN
 				break;
 #endif
 
+			case 'm':
+				monhost_child_prev = monhost_child_current;
```

Comment:
Starting value of "current" as declared on top is not "NULL". Care was taken to predefine others :)